CONTRIBUTING.rst

```diff
@@ -33,6 +33,70 @@ you are solving it. This might save you a lot of time if the maintainers
 are already working on it or have a specific idea on how the problem should
 be solved.
 
+Setting up the development environment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are three python tools required to develop, test and release ics.py:
+`poetry <https://python-poetry.org/>`_ for managing virtualenvs plus dependencies and for building plus publishing the package,
+`tox <https://tox.readthedocs.io/>`_ for running the testsuite and building the documentation,
+and `bumpversion <https://pypi.org/project/bumpversion/>`_ to help with making a release.
+Their respective configuration files are ``pyproject.toml``, ``tox.ini`` and ``.bumpversion.cfg``.
+The ``poetry.lock`` file optionally locks the dependency versions against which we want to develop,
+which is independent from the versions the library pulls in when installed as a dependency itself (where we are pretty liberal),
+and the versions we test against (which is always the latest releases installed by tox).
+You can simply install the tools via pip:
+
+.. code-block:: bash
+
+    $ pip install tox poetry bumpversion --user
+
+.. note::
+    If you want to develop using multiple different python versions, you might want to consider the
+    `poetry installer <https://python-poetry.org/docs/#installation>`_.
+
+    Poetry will automatically manage a virtualenv that you can use for developing.
+    By default, it will be located centrally in your home directory (e.g. in ``/home/user/.cache/pypoetry/virtualenvs/``).
+    To make poetry use a ``./.venv/`` directory within the ics.py folder use the following config:
+
+    .. code-block:: bash
+
+        $ poetry config virtualenvs.in-project true
+
+Now you are ready to setup your development environment using the following command:
+
+.. code-block:: bash
+
+    $ poetry install
+
+This will create a new virtualenv and install the dependencies for using ics.py.
+Furthermore, the current source of the ics.py package will be available similar to running ``./setup.py develop``.
+To access the virtualenv, simply use ``poetry run python`` or ``poetry shell``.
+If you made some changes and now want to lint your code, run the testsuite, or build the documentation, simply call tox.
+You don't have to worry about which versions in which venvs are installed and whether you're directly testing against the sources or against a built package, tox handles all that for you:
+
+.. code-block:: bash
+
+    $ tox
+
+To just run a single task and not the whole testsuite, use the ``-e`` flag:
+
+.. code-block:: bash
+
+    $ tox -e docs
+
+Checkout ``tox.ini`` to see which tasks are available.
```

Comment:
IMHO we should avoid this advise. This can be done with tox itself. I would recommend to use `tox-av`. As such I would rewrite this paragraph as:

```suggestion
To get a list of all available tasks, run :command:`tox -av`.
```